src/selfplay/loop.cc

```diff
@@ -123,13 +126,46 @@ void SelfPlayLoop::CmdSetOption(const std::string& name,
 }
 
 void SelfPlayLoop::SendTournament(const TournamentInfo& info) {
+  int winp1   = info.results[0][0] + info.results[0][1];
+  int loosep1 = info.results[2][0] + info.results[2][1];
+  int draws   = info.results[1][0] + info.results[1][1];
+  float perct = -1, elo = 99999;
+  float los   = 99999;
+  if ((winp1 + loosep1 + draws) > 0)
+    perct = (((float)draws) / 2 + winp1) / (winp1 + loosep1 + draws);
+  if ((perct < 1) && (perct > 0)) elo = -400 * log(1 / perct - 1) / log(10);
+  if ((winp1 + loosep1) > 0)
+    los = .5 +
+          .5 * std::erf((winp1 - loosep1) / std::sqrt(2.0 * (winp1 + loosep1)));
+
   std::string res = "tournamentstatus";
   if (info.finished) res += " final";
-  res += " win " + std::to_string(info.results[0][0]) + " " +
-         std::to_string(info.results[0][1]);
-  res += " lose " + std::to_string(info.results[2][0]) + " " +
-         std::to_string(info.results[2][1]);
-  res += " draw " + std::to_string(info.results[1][0]) + " " +
+  res += " P1: +" + std::to_string(winp1) + " -" + std::to_string(loosep1) +
+         " =" + std::to_string(draws);
+
+  if (perct > 0) {
+    std::ostringstream oss;
+    oss << std::fixed << std::setw(5) << std::setprecision(2) << (perct * 100)
+        << "%";
+    res += " Win: " + oss.str();
+  }
+  if (elo < 99998) {
```

Comment:
What are those magic constants? I don't understand what's going on here, but looks like a hack.

Reply:
Its just for the case where no elo can be computed (its preinitialized to 99999). There are a number of cases where that can happen (all wins, all losses, no games etc.). One could of course check for all these cases, but that would also look bloated. The number is arbitrary.